src/components/search/SearchService.js

```diff
@@ -10,6 +10,7 @@ goog.provide('ga_search_service');
     '[0-9]+\\b)("|\'\'|′′|″)';
   var DMSNorth = '[N]';
   var DMSEast = '[E]';
+  var MGRS = '3[123]\[\\s\a-z]{3}[\\s\\d]*';
```

Comment:
Please assure that start of string is 3 (adding ^). Otherwise, this looks good to me.